projects/opendr_ws_2/src/opendr_perception/opendr_perception/object_tracking_2d_deep_sort_node.py

```diff
@@ -0,0 +1,246 @@
+
+        # Convert detected boxes to ROS type and publish
+        ros_boxes = self.bridge.to_ros_boxes(detection_boxes)
+        if self.detection_publisher is not None:
+            self.detection_publisher.publish(ros_boxes)
+            self.get_logger().info("Published " + str(len(detection_boxes)) + " detection boxes")
+
+        ros_ids = Int32MultiArray()
+        ros_ids.data = ids
+
+        if self.tracking_id_publisher is not None:
+            self.tracking_id_publisher.publish(ros_ids)
+            self.get_logger().info("Published " + str(len(ids)) + " tracking ids")
+
+
+colors = [
+    (255, 0, 255),
+    (0, 0, 255),
+    (0, 255, 0),
+    (255, 0, 0),
+    (35, 69, 55),
+    (43, 63, 54),
+]
+
+
+def draw_predictions(frame, predictions: TrackingAnnotation, is_centered=False, is_flipped_xy=True):
+    global colors
+    w, h, _ = frame.shape
+
+    for prediction in predictions.boxes:
+        prediction = prediction
+
+        if not hasattr(prediction, "id"):
+            prediction.id = 0
+
+        color = colors[int(prediction.id) * 7 % len(colors)]
+
+        x = prediction.left
+        y = prediction.top
+
+        if is_flipped_xy:
+            x = prediction.top
+            y = prediction.left
+
+        if is_centered:
+            x -= prediction.width
+            y -= prediction.height
+
+        cv2.rectangle(
+            frame,
+            (int(x), int(y)),
+            (
+                int(x + prediction.width),
+                int(y + prediction.height),
+            ),
+            color,
+            2,
+        )
+
+
+def main(
+    args=None,
+):
+    rclpy.init(args=args)
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-n", "--model_name", help="Name of the trained model",
+                        type=str, default="deep_sort")
+    parser.add_argument("-t", "--temp_dir", help="Path to a temp dir with models",
+                        type=str, default="temp")
+    parser.add_argument("-i", "--input_image_topic",
+                        help="Input Image topic provdied by either an image_dataset_node, webcam or any other image node",
+                        type=str, default="/opendr/dataset_image")
+    parser.add_argument("-od", "--output_detection_topic",
+                        help="Output detections topic",
+                        type=str, default="/opendr/detection")
+    parser.add_argument("-ot", "--output_tracking_id_topic",
+                        help="Output detections topic",
+                        type=str, default="/opendr/tracking_id")
+    parser.add_argument("-oi", "--output_image_topic",
+                        help="Output detections topic",
+                        type=str, default="/opendr/image_annotated")
+    parser.add_argument("--device", help="Device to use, either \"cpu\" or \"cuda\", defaults to \"cuda\"",
+                        type=str, default="cuda", choices=["cuda", "cpu"])
+    args = parser.parse_args()
```

Comment:
As a reminder, whatever changes are made in the default values in the ctor above, also apply here.